pkg/controller/config/config_controller.go

```diff
@@ -398,16 +522,19 @@ func createCR(c client.Client) error {
 	return err
 }
 
-func isUpToDate(cfg *op.Config) bool {
+func stateCheck(cfg *op.Config, code op.InstallStatus) bool {
+	if !matchesUUID(cfg.Status.OperatorUUID) {
```

Comment:
this need not be checked for all states. If you follow the state machine above, we can move the stateCheck and the version check to just `Installed` state.